cmd/oras/internal/errors/errors.go

```diff
@@ -60,6 +65,45 @@ func CheckArgs(checker func(args []string) (bool, string), Usage string) cobra.P
 	}
 }
 
+// Handler handles error during cmd execution.
+type Handler interface {
+	// Handle handles error during cmd execution.
+	// if nil is returned, then err cannot be handled.
+	Handle(err error, cmd *cobra.Command) error
+}
+
+// Command returns an error-handled cobra command.
+func Command(cmd *cobra.Command, handler Handler) *cobra.Command {
+	runE := cmd.RunE
+	cmd.RunE = func(cmd *cobra.Command, args []string) error {
+		err := runE(cmd, args)
+		if err == nil {
+			return nil
+		}
+		if handled := handler.Handle(err, cmd); handled != nil {
+			return handled
+		}
+		return err
+	}
+	return cmd
+}
+
+// GetInner gets the inner error from the error response.
+func GetInner(err error, errResp *errcode.ErrorResponse) error {
+	inner := errResp.Errors
+	if len(inner) == 0 {
+		return fmt.Errorf("empty response body: %w", errResp)
+	} else {
+		errContent := err.Error()
+		errRespContent := errResp.Error()
+		if idx := strings.Index(errContent, errRespContent); idx > 0 {
+			// remove HTTP related info
+			return fmt.Errorf("%s%w", errContent[:idx], inner)
+		}
```

Comment:
Will it be safer to use `strings.Replace`?

Reply:
`strings.Replace` cannot exclude the case when `idx == 0`. Also it requires extra code to do error wrapping.